a10_octavia/controller/worker/tasks/vthunder_tasks.py

```diff
@@ -180,6 +180,19 @@ def execute(self, subnet, amphora, lb_count_flavor, lb_count_subnet, flavor_data
                                 subnet.network_id, amp)
 
 
+class UpdateL2DSR(VThunderBaseTask):
+    """Task to update wildcat address in allowed_address_pair for L2DSR"""
+
+    def execute(self, subnet, amphora, lb_count_subnet):
+        if CONF.vthunder.l2dsr_support:
+            for amp in amphora:
+                self.network_driver.allow_use_any_source_ip_on_egress(subnet.network_id, amp)
+        if not CONF.vthunder.l2dsr_support:
```

Comment:
I think we can't just remove the aap when l2dsr_support is not configured. Since the app may be added by flaovr (e.g. "deployment": {"dsr_type": "l2dsr_transparent"}) not configuration.

So, we also need to traverse flavor of loadbalancers that bind to the subnet. And make sure there is no l2dsr flavor before we remove the aap.

Reply:
Done